tracing-appender/src/file_appender.rs

```diff
@@ -0,0 +1,125 @@
+use crate::inner::BufWriterFactory;
+use crate::worker::Worker;
+use crate::Rotation;
+use chrono::Utc;
+use crossbeam_channel::{bounded, Sender};
+use std::path::Path;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+use std::{io, thread};
+use tracing_subscriber::fmt::MakeWriter;
+
+#[allow(dead_code)]
+pub struct FileAppender {
+    log_writer: FileWriter,
+    worker_thread: thread::JoinHandle<()>,
+    error_counter: Arc<AtomicU64>,
+}
+
+impl FileAppender {
+    pub fn new(log_directory: &str, log_filename_prefix: &str) -> Self {
+        FileAppenderBuilder::default()
+            .build(log_directory, log_filename_prefix)
+            .expect("Failed to create FileAppender")
+    }
+
+    fn create_appender(
+        sender: Sender<Vec<u8>>,
+        worker: Worker<BufWriterFactory>,
+        error_counter: Arc<AtomicU64>,
+    ) -> Self {
+        Self {
+            log_writer: FileWriter {
+                channel: sender,
+                error_counter: error_counter.clone(),
+            },
+            worker_thread: worker.worker_thread(),
+            error_counter,
+        }
+    }
+
+    pub fn builder() -> FileAppenderBuilder {
+        FileAppenderBuilder::default()
+    }
+
+    pub fn get_writer(self) -> FileWriter {
+        self.log_writer
+    }
+
+    pub fn get_error_counter(self) -> Arc<AtomicU64> {
+        self.error_counter.clone()
+    }
```

Comment:
Since these both take `self` by value, it's not possible to get the `log_writer` _and_ the error counter out of a `FileAppender` after building it. Since we're cloning the `Arc` anyway, they should probably both take `&self` and clone the writer or error counter

Reply:
Sorry about that. Should have realized that would cause problems later. Still trying to get out of a Java mindset lol.